src/coreComponents/linearAlgebra/solvers/DirectSolver.cpp

```diff
@@ -0,0 +1,303 @@
+    pcols = num_procs/prows;
+  }
+  gridinfo_t grid;
+  superlu_gridinit( comm, prows, pcols, &grid );
+
+  // Call the linear equation solver.
+  int nrhs = 0;
+  int const ldb = b.localSize();
+  dSOLVEstruct_t SOLVEstruct;
+  array1d< real64 > berr( b.localSize() );
+  int info = 0;
+
+  options.Fact = DOFACT;
+  pdgssvx( &options,
+           &SLUDMat,
+           &ScalePermstruct,
+           NULL,
+           ldb,
+           nrhs,
+           &grid,
+           &LUstruct,
+           &SOLVEstruct,
+           berr.data(),
+           &stat,
+           &info );
+
+  timeFact = watch.elapsedTime();
+  watch.zero();
+
+  if( info == 0 )
+  {
+    options.Fact = FACTORED;
+    nrhs = 1;
+    x.copy( b );
+    pdgssvx( &options,
+             &SLUDMat,
+             &ScalePermstruct,
+             x.extractLocalVector(),
+             ldb,
+             nrhs,
+             &grid,
+             &LUstruct,
+             &SOLVEstruct,
+             berr.data(),
+             &stat,
+             &info );
+  }
+
+  timeSolve = watch.elapsedTime();
+
+  if( logLevel > 0 )
+  {
+    // Print the statistics.
+    PStatPrint( &options, &stat, &grid );
+  }
+
+  // Deallocate other SuperLU data structures
+  dScalePermstructFree( &ScalePermstruct );
+  dDestroy_LU( n, &grid, &LUstruct );
+  dLUstructFree( &LUstruct );
+  if( options.SolveInitialized )
+  {
+    dSolveFinalize( &options, &SOLVEstruct );
+  }
+
+  return info;
+}
+
+template< typename LAI >
+void DirectSolver< LAI >::solve( Matrix & matrix,
+                                 Vector & b,
+                                 Vector & x ) const
+{
+  GEOSX_LOG_RANK_0( "===========> Using direct solver from SuperLU_Dist package. <===========" );
+  // Convert matrix from Matrix to SuperMatrix format
+  array1d< globalIndex > rowPtr;
+  array1d< globalIndex > cols;
+  array1d< real64 > vals;
+  SuperMatrix SLUDMat;
+  ConvertToSuperMatrix< LAI >( matrix, rowPtr, cols, vals, SLUDMat );
+
+  MPI_Comm const comm = matrix.getComm();
+
+  // Initialize options.
+  superlu_dist_options_t options;
+  set_default_options_dist( &options );
+  options.ReplaceTinyPivot = YES;
+  if( m_params.logLevel > 1 )
+  {
+    options.PrintStat = YES;
+  }
+  else
+  {
+    options.PrintStat = NO;
+  }
+
+  if( m_params.logLevel > 0 )
+  {
+    print_sp_ienv_dist( &options );
+    print_options_dist( &options );
+  }
+
+  real64 const bnorm2 = b.norm2();
+
+  real64 timeFact, timeSolve;
+  int const info = SolveSuperMatrix< LAI >( SLUDMat, b, x, comm, options, m_params.logLevel, timeFact, timeSolve );
+
+  Vector res( b );
+  matrix.gemv( -1.0, x, 1.0, res );
+  m_result.residualReduction = res.norm2() / bnorm2;
+
+  if( info == 0 && m_result.residualReduction < m_params.krylov.relTolerance )
+  {
+    m_result.status = LinearSolverResult::Status::Success;
+    m_result.setupTime = timeFact;
+    m_result.solveTime = timeSolve;
+  }
+  else
+  {
+    m_result.status = LinearSolverResult::Status::NotConverged;
+  }
```

Comment:
Move this check in `void HypreSolver::solve_direct` instead of

https://github.com/GEOSX/GEOSX/blob/eb241464c30398e1a8ee2617058ffd32d550fb79/src/coreComponents/linearAlgebra/interfaces/hypre/HypreSolver.cpp#L88-L90

Reply:
Done!